File: chroma-league/src/main/java/net/booone/chromaleague/hud/parts/Background.java
package net.booone.chromaleague.hud.parts;

import net.booone.chromaleague.hud.animations.BaronBuffBackgroundAnimation;
import net.booone.chromaleague.hud.colors.BackgroundBreathingColor;
import net.booone.chromaleague.rest.gamestats.MapTerrain;
import net.booone.chromaleague.state.GameStateHelper;
import net.booone.chromaleague.state.RunningState;
import net.booone.razersdk.animation.AnimatedFrame;
import net.booone.razersdk.animation.LayeredFrame;
import net.booone.razersdk.animation.SimpleFrame;
import net.booone.razersdk.color.BreathingColor;
import net.booone.razersdk.color.Color;
import net.booone.razersdk.color.StaticColor;

public class Background extends LayeredFrame {
    public static final StaticColor DEFAULT_BACKGROUND_COLOR = new StaticColor(5, 5, 5);
    private static final AnimatedFrame BARON_BUFF_BACKGROUND_ANIMATION = new BaronBuffBackgroundAnimation();
    private static final BreathingColor DEAD_BACKGROUND = new BackgroundBreathingColor(new StaticColor(60, 40, 40));

    public Background() {
        addFrame(new SimpleFrame(getCurrentBackgroundColor()));
        if (GameStateHelper.hasBaronBuff()) {
            addFrame(BARON_BUFF_BACKGROUND_ANIMATION);
        }
    }

    public static Color getCurrentBackgroundColor() {
        if (GameStateHelper.isActivePlayerAlive()) {
            if (RunningState.getGameState().getEventData().didRiftAnimationPlay()) {
                return MapTerrain.fromApiType(RunningState.getGameState().getGameStats().mapTerrain()).getBackgroundColor();
            } else {
                return DEFAULT_BACKGROUND_COLOR;
            }
        }
        return DEAD_BACKGROUND.getColor();
    }
}

File: src/hud/parts/Background.ts
import {AnimatedFrame} from '../../razer-sdk/animation/AnimatedFrame.js';
import {LayeredFrame} from '../../razer-sdk/animation/LayeredFrame.js';
import {SimpleFrame} from '../../razer-sdk/animation/SimpleFrame.js';
import {BreathingColor} from '../../razer-sdk/color/BreathingColor.js';
import {Color} from '../../razer-sdk/color/Color.js';
import {StaticColor} from '../../razer-sdk/color/StaticColor.js';
import {mapTerrainFromApiType, mapTerrainGetBackgroundColor} from '../../rest/gamestats/MapTerrain.js';
import * as GameStateHelper from '../../state/GameStateHelper.js';
import {RunningState} from '../../state/RunningState.js';
import {BaronBuffBackgroundAnimation} from '../animations/BaronBuffBackgroundAnimation.js';
import {BackgroundBreathingColor} from '../colors/BackgroundBreathingColor.js';
import {DEFAULT_BACKGROUND_COLOR} from './BackgroundConstants.js';

export { DEFAULT_BACKGROUND_COLOR };

const BARON_BUFF_BACKGROUND_ANIMATION: AnimatedFrame = new BaronBuffBackgroundAnimation();
const DEAD_BACKGROUND: BreathingColor = new BackgroundBreathingColor(new StaticColor(60, 40, 40));

export class Background extends LayeredFrame {
    constructor() {
        super();
        this.addFrame(new SimpleFrame(getCurrentBackgroundColor()));
        if (GameStateHelper.hasBaronBuff()) {
            this.addFrame(BARON_BUFF_BACKGROUND_ANIMATION);
        }
    }
}

export function getCurrentBackgroundColor(): Color {
    if (GameStateHelper.isActivePlayerAlive()) {
        if (RunningState.getGameState().eventData.riftAnimationPlayed) {
            return mapTerrainGetBackgroundColor(
                mapTerrainFromApiType(RunningState.getGameState().gameStats!.mapTerrain),
            );
        } else {
            return DEFAULT_BACKGROUND_COLOR;
        }
    }
    return DEAD_BACKGROUND.getColor();
}
